staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -344,6 +385,24 @@ type MatchResources struct {
 	MatchPolicy *MatchPolicyType `json:"matchPolicy,omitempty" protobuf:"bytes,7,opt,name=matchPolicy,casttype=MatchPolicyType"`
 }
 
+// ValidationAction specifies a policy enforcement action.
+// +enum
+type ValidationAction string
+
+const (
+	// Deny specifies that a validation failure results in a denied request.
+	Deny ValidationAction = "Deny"
+	// Warn specifies that a validation failure is reported to the request client
+	// in HTTP Warning headers, with a warning code of 299. Warnings can be sent
+	// both for allowed or denied admission responses.
+	Warn ValidationAction = "Warn"
```

Comment:
I'm not sure the combination of Deny+Warn makes sense. Consider whether to just doc this or prevent in validation

Deny+Audit and Warn+Audit make sense.

Reply:
Added validation and doc